Python program throwing error about a type mismatch but the type seems correct

### What happened?

Using Python with a TF module and getting the error below about expecting a `str` for `peerings` but getting a `mapping` when in fact mapping is correct and is what the generated SDK expects.

Here is the `ModuleArgs` definition that was generated:
```
class ModuleArgs(
    *,
    address_space: Input[Sequence[Input[str]]] | None = None,
    bgp_community: Input[str] | None = None,
    ddos_protection_plan: Input[DdosProtectionPlanArgs] | None = None,
    diagnostic_settings: Input[Mapping[str, Input[DiagnosticSettingsArgs]]] | None = None,
    dns_servers: Input[DnsServersArgs] | None = None,
    enable_telemetry: Input[bool] | None = None,
    enable_vm_protection: Input[bool] | None = None,
    encryption: Input[EncryptionArgs] | None = None,
    extended_location: Input[ExtendedLocationArgs] | None = None,
    flow_timeout_in_minutes: Input[float] | None = None,
    location: Input[str] | None = None,
    lock: Input[LockArgs] | None = None,
    name: Input[str] | None = None,
    peerings: Input[Mapping[str, Input[PeeringsArgs]]] | None = None,
    resource_group_name: Input[str] | None = None,
    role_assignments: Input[Mapping[str, Input[RoleAssignmentsArgs]]] | None = None,
    subnets: Input[Mapping[str, Input[SubnetsArgs]]] | None = None,
    subscription_id: Input[str] | None = None,
    tags: Input[Mapping[str, Input[str]]] | None = None
)
```
The docs for the module (https://registry.terraform.io/modules/Azure/avm-res-network-virtualnetwork/azurerm/latest?tab=inputs) show peerings to be an optional map.

Error Message:
```
Diagnostics:
  pulumi:pulumi:Stack (azure-tf-module-py-dev):
    error: Program failed with an unhandled exception:
    Traceback (most recent call last):
      File "/Users/mitch/.pulumi/bin/pulumi-language-python-exec", line 198, in <module>
        loop.run_until_complete(coro)
      File "/Users/mitch/.pyenv/versions/3.11.8/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
        return future.result()
               ^^^^^^^^^^^^^^^
      File "/Users/mitch/Downloads/test-pulumi-tf-module/azure-tf-module-py/venv/lib/python3.11/site-packages/pulumi/runtime/stack.py", line 143, in run_in_stack
        await run_pulumi_func(run)
      File "/Users/mitch/Downloads/test-pulumi-tf-module/azure-tf-module-py/venv/lib/python3.11/site-packages/pulumi/runtime/stack.py", line 55, in run_pulumi_func
        await wait_for_rpcs()
      File "/Users/mitch/Downloads/test-pulumi-tf-module/azure-tf-module-py/venv/lib/python3.11/site-packages/pulumi/runtime/stack.py", line 119, in wait_for_rpcs
        await task
      File "/Users/mitch/Downloads/test-pulumi-tf-module/azure-tf-module-py/venv/lib/python3.11/site-packages/pulumi/runtime/resource.py", line 1105, in do_register
        rpc.resolve_outputs(
      File "/Users/mitch/Downloads/test-pulumi-tf-module/azure-tf-module-py/venv/lib/python3.11/site-packages/pulumi/runtime/rpc.py", line 1414, in resolve_outputs
        translated_value = translate_output_properties(
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/Users/mitch/Downloads/test-pulumi-tf-module/azure-tf-module-py/venv/lib/python3.11/site-packages/pulumi/runtime/rpc.py", line 1269, in translate_output_properties
        raise AssertionError(
    AssertionError: Unexpected type; expected a value of type `<class 'str'>` but got a value of type `<class 'dict'>` at resource `vnet`, property `peerings`: {}
```




### Example

Run the program below and see the error.
You can uncomment the `peerings` property and it appears to get past the error, but then an additional similar "expected str got dict" type of error pops up.

```
"""
An Pulumi python program that leverages Terraform modules from https://azure.github.io/Azure-Verified-Modules/

pulumi package add terraform-module Azure/avm-res-network-virtualnetwork/azurerm 0.8.1 vnet 
"""

import pulumi
import pulumi_azure as azure
from pulumi_azure import core, storage
import pulumi_vnet as vnet

from netaddr import IPNetwork

config = pulumi.Config()
network_cidr = config.get('network_cidr', '10.0.0.0/24') 
subnet_prefix = config.get_int('subnet_prefix', 26)

# Create an Azure Resource Group
resource_group = core.ResourceGroup('resource_group')

# Create a Virtual Network with multiple subnets using imported terraform module
# Use python netaddr package to generate the subnet addresses and build the subnet list for the vnet resource
network = IPNetwork(network_cidr)
subnet_ips = network.subnet(subnet_prefix)
subnet_cidrs = [str(ip) for ip in subnet_ips]
subnets = {}
for subnet_cidr in subnet_cidrs:
    subnet_name=f"subnet-{subnet_cidr}"
    subnets[subnet_name] = vnet.SubnetsArgs(
        name=subnet_name,
        address_prefix=subnet_cidr
    )

# vnet_provider = vnet.Provider('vnet_provider', 
#     azurerm={"features":azure.ProviderFeaturesArgs(      
#         resource_group=azure.ProviderFeaturesResourceGroupArgs(prevent_deletion_if_contains_resources = False))
#     }
# )

vnet_provider = vnet.Provider('vnet_provider', 
    azurerm={"features": {"resource_group": {"prevent_deletion_if_contains_resources": False}}}
)

vnet = vnet.Module('vnet', vnet.ModuleArgs(
                    resource_group_name=resource_group.name,
                    location=resource_group.location,
                    name='vnet',
                    address_space=['10.0.0.0/16'],
                    subnets=subnets
                    #     peerings={"vnet-peering": vnet.PeeringsArgs(
                    #         name='vnet-peering',
                    #         remote_virtual_network_resource_id='fakevpc',
                    #         allow_virtual_network_access=True,)}
                    #    
                   ),
                   opts=pulumi.ResourceOptions(provider=vnet_provider)
)

pulumi.export('resourcegroup_name', resource_group.name)
```

### Output of `pulumi about`

CLI          
Version      3.156.0
Go Version   go1.24.1
Go Compiler  gc

Plugins
KIND      NAME              VERSION
resource  azure             6.21.0
language  python            3.156.0
resource  terraform-module  0.0.5
resource  terraform-module  0.0.5

Host     
OS       darwin
Version  15.3.1
Arch     x86_64

This project is written in python: executable='/Users/mitch/Downloads/test-pulumi-tf-module/azure-tf-module-py/venv/bin/python' version='3.11.8'

Current Stack: demo/azure-tf-module-py/dev

TYPE                                    URN
pulumi:pulumi:Stack                     urn:pulumi:dev::azure-tf-module-py::pulumi:pulumi:Stack::azure-tf-module-py-dev
pulumi:providers:azure                  urn:pulumi:dev::azure-tf-module-py::pulumi:providers:azure::default_6_21_0
pulumi:providers:azure                  urn:pulumi:dev::azure-tf-module-py::pulumi:providers:azure::azureprovider
pulumi:providers:vnet                   urn:pulumi:dev::azure-tf-module-py::pulumi:providers:vnet::vnet_provider
azure:core/resourceGroup:ResourceGroup  urn:pulumi:dev::azure-tf-module-py::azure:core/resourceGroup:ResourceGroup::resource_group


Found no pending operations associated with demo/dev

Backend        
Name           pulumi.com

Dependencies:
NAME          VERSION
netaddr       1.3.0
pulumi_aks    9.4.1
pulumi_azure  6.21.0
pulumi_vnet   0.8.1
setuptools    76.0.0
wheel         0.45.1

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/pkg/modprovider/module_schema_overrides/cloudposse-label.json
+++ b/pkg/modprovider/module_schema_overrides/cloudposse-label.json
@@ -1,0 +1,62 @@
+{
+    "source": "cloudposse/label/null",
+    "maximumVersion": "1.0.0",
+    "partialSchema": {
+      "outputs": {
+        "enabled": {
+          "type": "boolean"
+        },
+        "attributes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "additional_tag_map": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "label_order": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "regex_replace_chars": {
+          "type": "string"
+        },
+        "id_length_limit": {
+          "type": "number"
+        },
+        "tags_as_list_of_maps": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "additionalProperties": {
+              "type": "string"
+            }
+          }
+        },
+        "descriptors": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "normalized_context": {
+          "$ref": "pulumi.json#/Any"
+        },
+        "context": {
+          "$ref": "pulumi.json#/Any"
+        }
+      }
+    }
+  }

--- a/pkg/modprovider/module_schema_overrides/terraform-aws-acm.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-aws-acm.json
@@ -1,0 +1,44 @@
+{
+    "source": "terraform-aws-modules/acm/aws",
+    "maximumVersion": "6.0.0",
+    "partialSchema": {
+        "outputs": {
+            "acm_certificate_domain_validation_options": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "additionalProperties": {
+                        "$ref": "pulumi.json#/Any"
+                    }
+                }
+            },
+            "acm_certificate_validation_emails": {
+                "type": "array",
+                "items": {
+                    "type": "string"
+                }
+            },
+            "distinct_domain_names": {
+                "type": "array",
+                "items": {
+                    "type": "string"
+                }
+            },
+            "validation_domains": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "additionalProperties": {
+                        "$ref": "pulumi.json#/Any"
+                    }
+                }
+            },
+            "validation_route53_record_fqdns": {
+                "type": "array",
+                "items": {
+                    "type": "string"
+                }
+            }
+        }
+    }
+}

--- a/pkg/modprovider/module_schema_overrides/terraform-aws-alb.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-aws-alb.json
@@ -1,0 +1,32 @@
+{
+  "source": "terraform-aws-modules/alb/aws",
+  "maximumVersion": "10.0.0",
+  "partialSchema": {
+    "outputs": {
+      "listeners": {
+        "type": "object",
+        "additionalProperties": {
+          "$ref": "pulumi.json#/Any"
+        }
+      },
+      "listener_rules": {
+        "type": "object",
+        "additionalProperties": {
+          "$ref": "pulumi.json#/Any"
+        }
+      },
+      "target_groups": {
+        "type": "object",
+        "additionalProperties": {
+          "$ref": "pulumi.json#/Any"
+        }
+      },
+      "route53_records": {
+        "type": "object",
+        "additionalProperties": {
+          "$ref": "pulumi.json#/Any"
+        }
+      }
+    }
+  }
+}

--- a/pkg/modprovider/module_schema_overrides/terraform-aws-eks.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-aws-eks.json
@@ -1,0 +1,155 @@
+{
+    "source": "terraform-aws-modules/eks/aws",
+    "maximumVersion": "21.0.0",
+    "partialSchema": {
+      "outputs": {
+        "cluster_arn": {
+          "type": "string"
+        },
+        "cluster_certificate_authority_data": {
+          "type": "string"
+        },
+        "cluster_endpoint": {
+          "type": "string"
+        },
+        "cluster_id": {
+          "type": "string"
+        },
+        "cluster_name": {
+          "type": "string"
+        },
+        "cluster_oidc_issuer_url": {
+          "type": "string"
+        },
+        "cluster_dualstack_oidc_issuer_url": {
+          "type": "string"
+        },
+        "cluster_version": {
+          "type": "string"
+        },
+        "cluster_platform_version": {
+          "type": "string"
+        },
+        "cluster_status": {
+          "type": "string"
+        },
+        "cluster_primary_security_group_id": {
+          "type": "string"
+        },
+        "cluster_service_cidr": {
+          "type": "string"
+        },
+        "cluster_ip_family": {
+          "type": "string"
+        },
+        "access_entries": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "access_policy_associations": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "kms_key_arn": {
+          "type": "string"
+        },
+        "kms_key_id": {
+          "type": "string"
+        },
+        "kms_key_policy": {
+          "type": "string"
+        },
+        "cluster_security_group_arn": {
+          "type": "string"
+        },
+        "cluster_security_group_id": {
+          "type": "string"
+        },
+        "node_security_group_arn": {
+          "type": "string"
+        },
+        "node_security_group_id": {
+          "type": "string"
+        },
+        "oidc_provider": {
+          "type": "string"
+        },
+        "oidc_provider_arn": {
+          "type": "string"
+        },
+        "cluster_tls_certificate_sha1_fingerprint": {
+          "type": "string"
+        },
+        "cluster_iam_role_name": {
+          "type": "string"
+        },
+        "cluster_iam_role_arn": {
+          "type": "string"
+        },
+        "cluster_iam_role_unique_id": {
+          "type": "string"
+        },
+        "node_iam_role_name": {
+          "type": "string"
+        },
+        "node_iam_role_arn": {
+          "type": "string"
+        },
+        "node_iam_role_unique_id": {
+          "type": "string"
+        },
+        "cluster_addons": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "cluster_identity_providers": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "cloudwatch_log_group_name": {
+          "type": "string"
+        },
+        "cloudwatch_log_group_arn": {
+          "type": "string"
+        },
+        "fargate_profiles": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "eks_managed_node_groups": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "eks_managed_node_groups_autoscaling_group_names": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "self_managed_node_groups": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "self_managed_node_groups_autoscaling_group_names": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    }
+  }

--- a/pkg/modprovider/module_schema_overrides/terraform-aws-kms.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-aws-kms.json
@@ -1,0 +1,38 @@
+{
+    "source": "terraform-aws-modules/kms/aws",
+    "maximumVersion": "4.0.0",
+    "partialSchema": {
+      "outputs": {
+        "key_arn": {
+          "type": "string"
+        },
+        "key_id": {
+          "type": "string"
+        },
+        "key_policy": {
+          "type": "string"
+        },
+        "external_key_expiration_model": {
+          "type": "string"
+        },
+        "external_key_state": {
+          "type": "string"
+        },
+        "external_key_usage": {
+          "type": "string"
+        },
+        "aliases": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "grants": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        }
+      }
+    }
+  }

--- a/pkg/modprovider/module_schema_overrides/terraform-aws-rds-aurora.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-aws-rds-aurora.json
@@ -1,0 +1,89 @@
+{
+    "source": "terraform-aws-modules/rds-aurora/aws",
+    "maximumVersion": "10.0.0",
+    "partialSchema": {
+      "outputs": {
+        "cluster_members": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "cluster_port": {
+          "type": "number"
+        },
+        "cluster_master_password": {
+          "type": "string"
+        },
+        "cluster_master_username": {
+          "type": "string"
+        },
+        "cluster_master_user_secret": {
+          "type": "string"
+        },
+        "cluster_hosted_zone_id": {
+          "type": "string"
+        },
+        "cluster_ca_certificate_identifier": {
+          "type": "string"
+        },
+        "cluster_ca_certificate_valid_till": {
+          "type": "string"
+        },
+        "cluster_instances": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "additional_cluster_endpoints": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "cluster_role_associations": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "enhanced_monitoring_iam_role_name": {
+          "type": "string"
+        },
+        "enhanced_monitoring_iam_role_arn": {
+          "type": "string"
+        },
+        "enhanced_monitoring_iam_role_unique_id": {
+          "type": "string"
+        },
+        "security_group_id": {
+          "type": "string"
+        },
+        "db_cluster_parameter_group_arn": {
+          "type": "string"
+        },
+        "db_cluster_parameter_group_id": {
+          "type": "string"
+        },
+        "db_parameter_group_arn": {
+          "type": "string"
+        },
+        "db_parameter_group_id": {
+          "type": "string"
+        },
+        "db_cluster_cloudwatch_log_groups": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "db_cluster_activity_stream_kinesis_stream_name": {
+          "type": "string"
+        },
+        "db_cluster_secretsmanager_secret_rotation_enabled": {
+          "type": "boolean"
+        }
+      }
+    }
+  }

--- a/pkg/modprovider/module_schema_overrides/terraform-aws-rds.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-aws-rds.json
@@ -1,0 +1,92 @@
+{
+    "source": "terraform-aws-modules/rds/aws",
+    "maximumVersion": "7.0.0",
+    "partialSchema": {
+      "outputs": {
+        "enhanced_monitoring_iam_role_name": {
+          "type": "string"
+        },
+        "enhanced_monitoring_iam_role_arn": {
+          "type": "string"
+        },
+        "db_instance_address": {
+          "type": "string"
+        },
+        "db_instance_arn": {
+          "type": "string"
+        },
+        "db_instance_availability_zone": {
+          "type": "string"
+        },
+        "db_instance_endpoint": {
+          "type": "string"
+        },
+        "db_listener_endpoint": {
+          "type": "string"
+        },
+        "db_instance_engine": {
+          "type": "string"
+        },
+        "db_instance_engine_version_actual": {
+          "type": "string"
+        },
+        "db_instance_hosted_zone_id": {
+          "type": "string"
+        },
+        "db_instance_identifier": {
+          "type": "string"
+        },
+        "db_instance_resource_id": {
+          "type": "string"
+        },
+        "db_instance_status": {
+          "type": "string"
+        },
+        "db_instance_name": {
+          "type": "string"
+        },
+        "db_instance_username": {
+          "type": "string"
+        },
+        "db_instance_port": {
+          "type": "string"
+        },
+        "db_instance_ca_cert_identifier": {
+          "type": "string"
+        },
+        "db_instance_domain": {
+          "type": "string"
+        },
+        "db_instance_domain_auth_secret_arn": {
+          "type": "string"
+        },
+        "db_instance_domain_dns_ips": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "db_instance_domain_fqdn": {
+          "type": "string"
+        },
+        "db_instance_domain_iam_role_name": {
+          "type": "string"
+        },
+        "db_instance_domain_ou": {
+          "type": "string"
+        },
+        "db_instance_master_user_secret_arn": {
+          "type": "string"
+        },
+        "db_instance_cloudwatch_log_groups": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "db_instance_secretsmanager_secret_rotation_enabled": {
+          "type": "boolean"
+        }
+      }
+    }
+  }

--- a/pkg/modprovider/module_schema_overrides/terraform-aws-s3-bucket.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-aws-s3-bucket.json
@@ -1,0 +1,44 @@
+{
+    "source": "terraform-aws-modules/s3-bucket/aws",
+    "maximumVersion": "5.0.0",
+    "partialSchema": {
+      "outputs": {
+        "s3_bucket_id": {
+          "type": "string"
+        },
+        "s3_bucket_arn": {
+          "type": "string"
+        },
+        "s3_bucket_bucket_domain_name": {
+          "type": "string"
+        },
+        "s3_bucket_bucket_regional_domain_name": {
+          "type": "string"
+        },
+        "s3_bucket_hosted_zone_id": {
+          "type": "string"
+        },
+        "s3_bucket_lifecycle_configuration_rules": {
+          "$ref": "pulumi.json#/Any"
+        },
+        "s3_bucket_policy": {
+          "type": "string"
+        },
+        "s3_bucket_region": {
+          "type": "string"
+        },
+        "s3_bucket_website_endpoint": {
+          "type": "string"
+        },
+        "s3_bucket_website_domain": {
+          "type": "string"
+        },
+        "s3_directory_bucket_name": {
+          "type": "string"
+        },
+        "s3_directory_bucket_arn": {
+          "type": "string"
+        }
+      }
+    }
+  }

--- a/pkg/modprovider/module_schema_overrides/terraform-aws-vpc.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-aws-vpc.json
@@ -2,6 +2,415 @@
     "source": "terraform-aws-modules/vpc/aws",
     "maximumVersion": "6.0.0",
     "partialSchema": {
-        "nonNilOutputs": ["vpc_id"]
+      "outputs": {
+        "vpc_enable_dns_support": {
+          "type": "boolean"
+        },
+        "vpc_enable_dns_hostnames": {
+          "type": "boolean"
+        },
+        "vpc_secondary_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "vpc_owner_id": {
+          "type": "string"
+        },
+        "vpc_block_public_access_exclusions": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "egress_only_internet_gateway_id": {
+          "type": "string"
+        },
+        "nat_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "nat_public_ips": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "natgw_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "cgw_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "cgw_arns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "this_customer_gateway": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "default_vpc_enable_dns_support": {
+          "type": "boolean"
+        },
+        "default_vpc_enable_dns_hostnames": {
+          "type": "boolean"
+        },
+        "private_subnets": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "private_subnet_arns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "private_subnets_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "private_subnets_ipv6_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "public_subnets": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "public_subnet_arns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "public_subnets_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "public_subnets_ipv6_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "outpost_subnets": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "outpost_subnet_arns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "outpost_subnets_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "outpost_subnets_ipv6_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "database_subnets": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "database_subnet_arns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "database_subnets_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "database_subnets_ipv6_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "database_subnet_group": {
+          "type": "string"
+        },
+        "database_subnet_group_name": {
+          "type": "string"
+        },
+        "redshift_subnets": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "redshift_subnet_arns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "redshift_subnets_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "redshift_subnets_ipv6_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "redshift_subnet_group": {
+          "type": "string"
+        },
+        "elasticache_subnets": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "elasticache_subnet_arns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "elasticache_subnets_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "elasticache_subnets_ipv6_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "elasticache_subnet_group": {
+          "type": "string"
+        },
+        "elasticache_subnet_group_name": {
+          "type": "string"
+        },
+        "intra_subnets": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "intra_subnet_arns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "intra_subnets_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "intra_subnets_ipv6_cidr_blocks": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "private_route_table_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "public_route_table_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "database_route_table_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "redshift_route_table_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "elasticache_route_table_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "intra_route_table_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "private_nat_gateway_route_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "public_internet_gateway_route_id": {
+          "type": "string"
+        },
+        "public_internet_gateway_ipv6_route_id": {
+          "type": "string"
+        },
+        "database_internet_gateway_route_id": {
+          "type": "string"
+        },
+        "database_nat_gateway_route_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "database_ipv6_egress_route_id": {
+          "type": "string"
+        },
+        "private_ipv6_egress_route_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "public_route_table_association_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "private_route_table_association_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "database_route_table_association_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "redshift_route_table_association_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "redshift_public_route_table_association_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "elasticache_route_table_association_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "intra_route_table_association_ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "public_network_acl_id": {
+          "type": "string"
+        },
+        "public_network_acl_arn": {
+          "type": "string"
+        },
+        "private_network_acl_id": {
+          "type": "string"
+        },
+        "private_network_acl_arn": {
+          "type": "string"
+        },
+        "outpost_network_acl_id": {
+          "type": "string"
+        },
+        "outpost_network_acl_arn": {
+          "type": "string"
+        },
+        "intra_network_acl_id": {
+          "type": "string"
+        },
+        "intra_network_acl_arn": {
+          "type": "string"
+        },
+        "database_network_acl_id": {
+          "type": "string"
+        },
+        "database_network_acl_arn": {
+          "type": "string"
+        },
+        "redshift_network_acl_id": {
+          "type": "string"
+        },
+        "redshift_network_acl_arn": {
+          "type": "string"
+        },
+        "elasticache_network_acl_id": {
+          "type": "string"
+        },
+        "elasticache_network_acl_arn": {
+          "type": "string"
+        },
+        "azs": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "name": {
+          "type": "string"
+        }
+      }
     }
-}
+  }

--- a/pkg/modprovider/module_schema_overrides/terraform-azurerm-virtualnetwork.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-azurerm-virtualnetwork.json
@@ -1,0 +1,69 @@
+{
+  "source": "Azure/avm-res-network-virtualnetwork/azurerm",
+  "maximumVersion": "1.0.0",
+  "partialSchema": {
+    "outputs": {
+      "peerings": {
+        "type": "object",
+        "additionalProperties": {
+          "$ref": "#/types/[packageName]:index:PeeringResource"
+        }
+      },
+      "subnets": {
+        "type": "object",
+        "additionalProperties": {
+          "$ref": "#/types/[packageName]:index:SubnetResource"
+        }
+      },
+      "resource": {
+        "type": "object",
+        "additionalProperties": {
+          "$ref": "pulumi.json#/Any"
+        }
+      }
+    },
+    "supportingTypes": {
+      "[packageName]:index:SubnetResource": {
+        "type": "object",
+        "properties": {
+          "application_gateway_ip_configuration_resource_id": {
+            "type": "string"
+          }, 
+          "name": {
+            "type": "string"
+          },
+          "resource": {
+            "type": "object",
+            "additionalProperties": {
+              "$ref": "pulumi.json#/Any"
+            }
+          },
+          "resource_id": {
+            "type": "string"
+          }
+        }
+      },
+      "[packageName]:index:PeeringResource": {
+        "type": "object",
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "The name of the peering resource"
+          },
+          "resource_id": {
+            "type": "string",
+            "description": "The resource ID of the peering resource"
+          },
+          "reverse_name": {
+            "type": "string",
+            "description": "The name of the reverse peering resource"
+          },
+          "reverse_resource_id": {
+            "type": "string",
+            "description": "The resource ID of the reverse peering resource"
+          }
+        }
+      }
+    }
+  }
+}

--- a/pkg/modprovider/module_schema_overrides/terraform-google-kubernetes-engine.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-google-kubernetes-engine.json
@@ -1,0 +1,68 @@
+{
+    "source": "terraform-google-modules/kubernetes-engine/google",
+    "maximumVersion": "37.0.0",
+    "partialSchema": {
+        "outputs": {
+            "dns_cache_enabled": {
+                "type": "boolean"
+            },
+            "horizontal_pod_autoscaling_enabled": {
+                "type": "boolean"
+            },
+            "http_load_balancing_enabled": {
+                "type": "boolean"
+            },
+            "identity_service_enabled": {
+                "type": "boolean"
+            },
+            "instance_group_urls": {
+                "type": "array",
+                "items": {
+                    "type": "string"
+                }
+            },
+            "intranode_visibility_enabled": {
+                "type": "boolean"
+            },
+            "master_authorized_networks_config": {
+                "type": "object",
+                "additionalProperties": {
+                    "$ref": "pulumi.json#/Any"
+                }
+            },
+            "mesh_certificates_config": {
+                "type": "object",
+                "additionalProperties": {
+                    "$ref": "pulumi.json#/Any"
+                }
+            },
+            "network_policy_enabled": {
+                "type": "boolean"
+            },
+            "node_pools_names": {
+                "type": "array",
+                "items": {
+                    "type": "string"
+                }
+            },
+            "node_pools_versions": {
+                "type": "object",
+                "additionalProperties": {
+                    "type": "string"
+                }
+            },
+            "secret_manager_addon_enabled": {
+                "type": "boolean"
+            },
+            "vertical_pod_autoscaling_enabled": {
+                "type": "boolean"
+            },
+            "zones": {
+                "type": "array",
+                "items": {
+                    "type": "string"
+                }
+            }
+        }
+    }
+}

--- a/pkg/modprovider/module_schema_overrides/terraform-google-network.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-google-network.json
@@ -1,0 +1,53 @@
+{
+  "source": "terraform-google-modules/network/google",
+  "maximumVersion": "11.0.0",
+  "partialSchema": {
+    "outputs": {
+      "network": {
+        "$ref": "pulumi.json#/Any"
+      },
+      "subnets": {
+        "type": "object",
+        "additionalProperties": {
+          "$ref": "pulumi.json#/Any"
+        }
+      },
+      "subnets_names": {
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
+      },
+      "subnets_ips": {
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
+      },
+      "subnets_regions": {
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
+      },
+      "route_names": {
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
+      },
+      "subnets_private_access": {
+        "$ref": "pulumi.json#/Any"
+      },
+      "subnets_flow_logs": {
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
+      },
+      "subnets_secondary_ranges": {
+        "$ref": "pulumi.json#/Any"
+      }
+    }
+  }
+}

--- a/pkg/modprovider/module_schema_overrides/terraform-google-project-factory.json
+++ b/pkg/modprovider/module_schema_overrides/terraform-google-project-factory.json
@@ -1,0 +1,74 @@
+{
+    "source": "terraform-google-modules/project-factory/google",
+    "maximumVersion": "19.0.0",
+    "partialSchema": {
+      "outputs": {
+        "project_name": {
+          "type": "string"
+        },
+        "project_id": {
+          "type": "string"
+        },
+        "project_number": {
+          "type": "string"
+        },
+        "domain": {
+          "type": "string"
+        },
+        "group_email": {
+          "type": "string"
+        },
+        "service_account_id": {
+          "type": "string"
+        },
+        "service_account_display_name": {
+          "type": "string"
+        },
+        "service_account_email": {
+          "type": "string"
+        },
+        "service_account_name": {
+          "type": "string"
+        },
+        "service_account_unique_id": {
+          "type": "string"
+        },
+        "project_bucket_self_link": {
+          "type": "string"
+        },
+        "project_bucket_url": {
+          "type": "string"
+        },
+        "api_s_account": {
+          "type": "string"
+        },
+        "api_s_account_fmt": {
+          "type": "string"
+        },
+        "enabled_apis": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "enabled_api_identities": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "budget_name": {
+          "type": "string"
+        },
+        "tag_bindings": {
+          "type": "array",
+          "items": {
+            "$ref": "pulumi.json#/Any"
+          }
+        },
+        "usage_report_export_bucket": {
+          "type": "string"
+        }
+      }
+    }
+}

--- a/pkg/modprovider/tfmodules.go
+++ b/pkg/modprovider/tfmodules.go
@@ -472,10 +472,12 @@ func combineInferredModuleSchema(
 
 		if input.TypeSpec.Items != nil {
 			existingInput.TypeSpec.Items = input.TypeSpec.Items
+			existingInput.TypeSpec.AdditionalProperties = nil
 		}
 
 		if input.TypeSpec.AdditionalProperties != nil {
 			existingInput.TypeSpec.AdditionalProperties = input.TypeSpec.AdditionalProperties
+			existingInput.TypeSpec.Items = nil
 		}
 	}
 
@@ -501,9 +503,11 @@ func combineInferredModuleSchema(
 		}
 		if output.TypeSpec.Items != nil {
 			existingOutput.TypeSpec.Items = output.TypeSpec.Items
+			existingOutput.TypeSpec.AdditionalProperties = nil
 		}
 		if output.TypeSpec.AdditionalProperties != nil {
 			existingOutput.TypeSpec.AdditionalProperties = output.TypeSpec.AdditionalProperties
+			existingOutput.TypeSpec.Items = nil
 		}
 	}
 

